common/os_calls.c

```diff
@@ -583,6 +586,17 @@ g_sck_local_socket(void)
 }
 
 /*****************************************************************************/
+int
+g_sck_vsock_socket(void)
+{
+#if defined(XRDP_ENABLE_VSOCK)
+    return socket(PF_VSOCK, SOCK_STREAM, 0);
+#else
+    return 0;
+#endif
```

Comment:
should be -1

Reply:
Yea, xrdp currently treats sockets < 1 as invalid.  This should be changed in many places.  I'll add an issue for this.

Reply:
I add #938 for this